### ISSUE TITLE
Convert '<a>' tags that contains '&amp;'

### ISSUE THREAD

**Issue by rcrivera** [werk85/node-html-to-text#69]
There is a problem when to converting an `<a>` tag that contains `&amp;` on its `href` value and its text. 
```
var htmlToText = require('html-to-text');
var text = htmlToText.fromString('<a href="https://www.mywebsite.com?param1=value1&amp;param2=value2">Click &amp; Go</a>');
console.log(text); // Outputs: 'Click param2=value2]'
```

**Comment by chesles** [comment]
This seems to stem from a bug in [helper.decodeHTMLEntities](https://github.com/werk85/node-html-to-text/blob/master/lib/helper.js#L46-L50) where text such as "Click & [some-url-with-a-semicolon&amp;]" matches the regex with `& [some-url-with-a-semicolon&amp;` as the entity. That is obviously not a known/valid entity, and the entire chunk gets replaced with the result of `String.fromCharCode(NaN)`, which is a null byte.

**Comment by mlegenhausen** [comment]
Should be fixed by #70. Closed for now.
